### ISSUE TITLE
Cannot merge truecolor logos/images - PNG (ImageMagik backend)

### ISSUE THREAD

**Issue by icdsoftware** [SimpleSoftwareIO/simple-qrcode#206]
When you merge a truecolor logo or an image - it is not working properly as the QRCode image produced by BaconQrCode is not a truecolor itself, so the merge results in a PNG file with less colors that are required to display properly the logo.

To resolve the issue I had to slightly modify the Image class constuctor (SimpleSoftwareIO\QrCode\Image)

 ```
 public function __construct($image)
	{
		$t = imagecreatefromstring($image);
		if(!imageistruecolor($t)){
			$width = imagesx($t);
			$height = imagesy($t);
			$this->image = imagecreatetruecolor(imagesx($t), imagesy($t));
			imagecopyresampled($this->image, $t, 0, 0, 0, 0, $width,$height, $width, $height);
		}
		else{
			$this->image = $t;
		}

	}
```

You might be able to find a better place to add this, may be into the SimpleSoftwareIO\QrCode\ImageMerge class where the actual merge is done. 

I actually found a pending pull request, that most probably will fix the issue as it does create a truecolor destination image.
https://github.com/SimpleSoftwareIO/simple-qrcode/pull/201/files

**Comment by trinhcanhphuc** [comment]
Thank you so much, you help me <3

**Comment by SimplyCorey** [comment]
Resolved with #201
